src/libostree/ostree-sysroot-deploy.c

```diff
@@ -2082,48 +2082,19 @@ install_deployment_kernel (OstreeSysroot   *sysroot,
   g_autofree char *options_key = ostree_kernel_args_to_string (kargs);
   ostree_bootconfig_parser_set (bootconfig, "options", options_key);
 
-  g_autoptr(GError) local_error = NULL;
-  GKeyFile *config = ostree_repo_get_config (repo);
-  gchar **read_values = g_key_file_get_string_list (config, "sysroot", "bls-append-except-default", NULL, &local_error);
-  /* We can ignore not found errors */
-  if (!read_values)
-    {
-      gboolean not_found = g_error_matches (local_error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_KEY_NOT_FOUND) || \
-                           g_error_matches (local_error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_GROUP_NOT_FOUND);
-      if (not_found)
-        {
-          g_clear_error (&local_error);
-        }
-      else
-        {
-          g_propagate_error (error, g_steal_pointer (&local_error));
-          return FALSE;
-        }
-    }
-
   /* Only append to this BLS config if:
    * - this is not the default deployment
    */
    /* If deployment was prepended, it is the new default */
   gboolean is_new_default = (ostree_deployment_get_index (deployment) == 0);
   gboolean allow_append = !is_new_default;
-  if (allow_append)
+  if (repo && allow_append)
```

Comment:
Why the `if (repo)`?  Having a repository is a required part of ostree here, there should be no situations where it's `NULL`.

(In Rust we'd have references, which are non-nullable by default)

If you're concerned about this, it'd probably be OK to drop a `g_assert (repo != NULL)` near the top of the function...oh.  Oh no.  Right, now I see:

```
  /* Need the repo to try and extract the versions for deployments.
   * But this is a "nice-to-have" for the bootloader UI, so failure
   * here is not fatal to the whole operation.  We just gracefully
   * fall back to the deployment index. */
  g_autoptr(OstreeRepo) repo = NULL;
  (void) ostree_sysroot_get_repo (self, &repo, cancellable, NULL);
```

That code just needs to change.

Reply:
Can you rebase on https://github.com/ostreedev/ostree/pull/2667 ?